internal/repository/osbuildenvconfig/osbuildenvconfig.go

```diff
@@ -0,0 +1,35 @@
+package osbuildenvconfig
+
+import (
+	"context"
+
+	_ "github.com/golang/mock/mockgen/model"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/project-flotta/osbuild-operator/api/v1alpha1"
+)
+
+//go:generate mockgen -package=osbuildenvconfig -destination=mock_osbuildenvconfig.go . Repository
+type Repository interface {
+	Read(ctx context.Context, name string) (*v1alpha1.OSBuildEnvConfig, error)
+	Update(ctx context.Context, osBuildEnvConfig *v1alpha1.OSBuildEnvConfig) error
```

Comment:
Did you consider using `Patch`? It should be "safer" than `Update`, because `Update` will fail when the edited resource isn't fresh (was modified concurrently) and patch will try to merge the resources. 
I am not saying `Patch` is better than `Update`, but they have a bit different applications and behavior.

Reply:
Done